### ISSUE TITLE
Error: Unexpected! Inserting syntax kind of PropertyDeclaration, but Parameter was inserted.

### ISSUE THREAD

**Issue by gp187** [dsherret/ts-simple-ast#57]
The simplest 

`const property = classDeclaration.addProperty({ isStatic: true, name: "prop", type: "string" });`

triggers this on latest version when ever a non-empty constructor is present

`
constructor(
        private serv: MyLocalService
    ) {}
`
if no constructor then it works. 

```
ERROR >  Error: Unexpected! Inserting syntax kind of PropertyDeclaration, but Parameter was inserted.
    at Object.getRangeFromArray (/lab/o0/cli2/node_modules/ts-simple-ast/dist/manipulation/getRangeFromArray.js:11:19)
    at Object.fillAndGetChildren (/lab/o0/cli2/node_modules/ts-simple-ast/dist/manipulation/fillAndGetChildren.js:5:42)
    at Object.insertIntoBracesOrSourceFileWithFillAndGetChildren (/lab/o0/cli2/node_modules/ts-simple-ast/dist/manipulation/insertIntoBracesOrSourceFileWithFillAndGetChildren.js:16:33)
    at ClassDeclaration.insertProperties (/lab/o0/cli2/node_modules/ts-simple-ast/dist/compiler/class/ClassDeclaration.js:167:31)
    at ClassDeclaration.addProperties (/lab/o0/cli2/node_modules/ts-simple-ast/dist/compiler/class/ClassDeclaration.js:136:21)
    at as (/lab/o0/cli2/dist/index.js:283:13)
    at Object.<anonymous> (/lab/o0/cli2/dist/index.js:286:1)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:427:7)
    at startup (bootstrap_node.js:151:9)
    at bootstrap_node.js:542:3
```

**Comment by gp187** [comment]
Also, the variables get declared after the constructor. Wouldn't it be better before it?

**Comment by dsherret** [comment]
Thanks, I'm able to reproduce this and I've fixed the problem. Sorry about that! It was related to a change in behaviour I made in #39. When fixing this, it also made me notice an issue related to inserting. I'll fix that tonight as part of this and then do a new patch release.

#26 will deal with member ordering when calling an `add` method. It would be nice if people could configure the member order rules they want.

**Comment by dsherret** [comment]
@gp187 this is fixed in 0.49.0 so I'm going to close this. Let me know if you encounter any issues.

--

You should be able to get the parameters properties from a constructor by filtering the parameters for ones that are parameter properties:

```typescript
const ctorProps = ctor.getParameters().filter(p => p.isParameterProperty())
// p.isParameterProperty() == p.getScope() != null || p.isReadonly()
```

Is that what you mean? Otherwise, from a class declaration you can get them by calling:

```typescript
const ctorProps = classDeclaration.getInstanceProperties().filter(p => p instanceof ParameterDeclaration) as ParameterDeclaration[];
```

--

Unfortunately, right now removing nodes isn't possible. Removing a constructor is a bit of an exception. I'm working on that slowly in #47 (it's the highest priority feature right now)
